sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightPlugin.kt

```diff
@@ -42,40 +41,42 @@ abstract class SqlDelightPlugin : Plugin<Project> {
   @get:Inject
   abstract val registry: ToolingModelBuilderRegistry
 
-  private lateinit var extension: SqlDelightExtension
-
   override fun apply(project: Project) {
     require(GradleVersion.current() >= GradleVersion.version(MIN_GRADLE_VERSION)) {
       "SQLDelight requires Gradle version $MIN_GRADLE_VERSION or greater."
     }
 
-    extension = project.extensions.create("sqldelight", SqlDelightExtension::class.java)
+    val extension = project.extensions.create("sqldelight", SqlDelightExtension::class.java).apply {
+      linkSqlite.convention(true)
+    }
 
-    val androidPluginHandler = { _: Plugin<*> ->
+    project.plugins.withType(AndroidBasePlugin::class.java) {
       android.set(true)
       project.afterEvaluate {
-        project.setupSqlDelightTasks(afterAndroid = true)
+        project.setupSqlDelightTasks(afterAndroid = true, extension)
       }
     }
-    project.plugins.withId("com.android.application", androidPluginHandler)
-    project.plugins.withId("com.android.library", androidPluginHandler)
-    project.plugins.withId("com.android.instantapp", androidPluginHandler)
-    project.plugins.withId("com.android.feature", androidPluginHandler)
-    project.plugins.withId("com.android.dynamic-feature", androidPluginHandler)
-
-    val kotlinPluginHandler = { _: Plugin<*> -> kotlin.set(true) }
-    project.plugins.withId("org.jetbrains.kotlin.multiplatform", kotlinPluginHandler)
-    project.plugins.withId("org.jetbrains.kotlin.android", kotlinPluginHandler)
-    project.plugins.withId("org.jetbrains.kotlin.jvm", kotlinPluginHandler)
-    project.plugins.withId("org.jetbrains.kotlin.js", kotlinPluginHandler)
-    project.plugins.withId("kotlin2js", kotlinPluginHandler)
+
+    project.plugins.withType(KotlinBasePlugin::class.java) {
```

Comment:
I should note that `KotlinBasePlugin` was only added in Kotlin 1.7.0, so if compatibility with older versions is needed, this should be reverted.

https://kotlinlang.org/docs/whatsnew17.html#updates-in-the-kotlin-gradle-plugin-api

Reply:
While I am fine with requiring Kotlin 1.7.0, could you split this change from the PR to merge the other Gradle plugin changes and we can discuss the Kotlin requirement in separate PR?

Reply:
I think its fine, I can't imagine folks upgrading SQLDelight to not also be upgrading kotlin